src/App.js

```diff
@@ -29,17 +30,17 @@ const App = (props) => {
     });
     const [arePermissionsReady, setArePermissionReady] = useState(false);
     const [hasSystems, setHasSystems] = useState();
+    const [hasSystemsReady, setHasSystemsReady] = useState();
```

Comment:
Isn't this redundant? Wouldn't checking if `hasSystems` is set be the same?

Reply:
Oh, this is similar to what we do with `isLoading` in other places. Maybe `isLoading` would be a better name for it.

Reply:
The reason this is needed is that we should be able differentiate the cases when there no registered systems and when the API request is being made.

Reply:
I would also vote for "isLoading", "systemsLoading" or similar naming. Can we change that?

Also, please set it to some default boolean value so that we strictly distinguish between true/false only.